File: leetcode/move_zero/3.js
//思路：
//设定一个临时变量k=0，遍历数组nums，将非零元素与之前的零元素进行交换，维护变量k的值。
var moveZero = function(nums){
    let k = 0;
    for(let i = 0; i<nums. length; i++){
        if(nums[i])
            if(k!=i){
                nums[k] = nums[i];
                nums[i] = 0;
                k++;
            }
            else{
                k++;
            }
    }
    return nums;
}

File: leetcode/two_sum/3.go
package two_sum


//js 动态弱类型语言
// twoSum undefined => function
// go 类型约束 现代的c语言
func twoSum(nums []int,
	target int) []int {
		//o(n2) => o(n)
		// 2 7 9
		// for 
		// 9-2 = 7
		// {7:0,} hashtable
		// json
		h := make(map[int]int)//hashtable
		// js for python 
		for i , value := range nums{
		if wanted, ok := h[value]; ok {
			return []int {wanted, i}
		}else{
			h[target-value] = i 
		}
		return nil
	}
	return nil
}